### DISCUSSION TITLE
Generating Prisma schema from Nexus definitions

### DISCUSSION THREAD

**Discussion by rdmchr** [graphql-nexus/nexus#1032] [General]
I just started learning Nexus and Prisma and I was wondering if I could generate my Prisma schemas using my Nexus definitions?
For me, this would make a lot of things easier since I would only need to write the code once.
I did find that nexus-prisma plugin, but it seems like that requires me to start off by writing a Prisma schema, which could then generate at least a bare-bone Nexus definition.
To me, this would mean, that I have to write similar code twice, but feel free to correct me if I'm wrong.

Basically, what I'm looking for is a package that does, what the current nexus-prisma plugin does, but in reverse.

**Comment by georgekrax** [comment]
I do not believe that something like this exists, or will ever be developed. Prisma requires you to be specific, and the reverse functionality you ask for, has a lot of cases where the data type can be very generic.

Don't you have a DB, somewhere from where the data comes from and introspect this?

**Reply by rdmchr** [reply]
> Don't you have a DB, somewhere from where the data comes from and introspect this?

I don't quite understand what you are talking about here. Prisma creates the relevant tables and does all the CRUD; I don't write any SQL statements myself currently. Nexus creates my GraphQL schema, so I don't have to write that either. I thought the nexus-prisma plugin just provides a bridge between these two so that I can use the Prisma schema as TypeScript types for my Nexus definitions.

Originally I thought if I have to write my schema for Prisma, why can't I just use that to generate my GraphQL schema as well. I have since learned a lot more, so I can understand your first point.

**Reply by georgekrax** [reply]
You can use the new `nexus-prisma` plugin. Just run `yarn prisma generate` and it will also generate the corresponding Nexus schema, but you will have to add the types. Follow the guides at their [documentation](https://nexus.prisma.io/usage). Just remember to add:
```prisma
generator nexusPrisma {
   provider = "nexus-prisma"
}
```
At the top of your `schema.prisma` file

**Reply by rdmchr** [reply]
Yeah, I already figured that out

**Reply by georgekrax** [reply]
Then, what is the issue?

**Reply by rdmchr** [reply]
Originally, I hoped, that I could generate the Prisma schema with my Nexus definitions, but as mentioned above, I have already discovered that there is no good way to make that conversion. So my original question is already out of the window.
But as mentioned above, I don't understand what you meant by this statement:

> Don't you have a DB, somewhere from where the data comes from and introspect this?

To provide further detail, yes I do have a database that has my data and I use Prisma to do CRUD. But what do you mean with "introspect this?"

**Reply by georgekrax** [reply]
I meant that if you have  DB you can introspect it with Prisma, and let Prisma generate a `schema.prisma` file depending on your DB tables. Why do you want to turn the Nexus schema into a Prisma schema? You already have generated the Nexus schema from the Prisma one, with the help of `nexus-prisma`.

**Reply by rdmchr** [reply]
Oh, I see. No, I'm currently using Prisma the other way around. I let Prisma generate my database tables. You said, that I can generate the Nexus schema. Are you talking about Queries and Mutations? No, right?

**Reply by georgekrax** [reply]
I am talking about `objectType`s. Still cannot figure why you want to generate a Prisma schema from a Nexus one, when you have already done the opposite successfully.

**Reply by rdmchr** [reply]
> Originally, I hoped, that I could generate the Prisma schema with my Nexus definitions, but as mentioned above, I have already discovered that there is no good way to make that conversion.

I opened this threat shortly after discovering NexusJS. I thought I already made it clear that my question was caused by missing knowledge. Furthermore, did I state that I spend the last days using/learning Nexus. To make it clear: **My question was in hindsight stupid. My question has been answered.**

Thank you for trying to help me. 👍

**Reply by georgekrax** [reply]
Oh okay, no worries
